app/controllers/APIController.scala

```diff
@@ -50,3 +138,14 @@ class APIController @Inject() (
     }
   }
 }
+
+case class ProductUpdateRequest(
+    name: Option[String],
+    quantity: Option[Int],
+    price: Option[Double]
+)
+
+object ProductUpdateRequest {
+  implicit val requestReads: Format[ProductUpdateRequest] =
```

Comment:
the naming is off. If should be requestFormat. This doesn't affect usage, but you want to not be confusing

Reply:
val renamed to requestFormat